### PR TITLE
 fixed: deviation from cocoa when calling setTitle:''

### PR DESCRIPTION
**Pull request by daboe01**
previously, CPPopUpButton added the empty string to the item array. Cocoa clears the title and sets objectValue to -1.
this PR makes CPPopUpButton behave as in cocoa.
fixes issue #2471

### CODE REVIEW

**Comment by cacaodev** [comment]
Looks good to me.
Do you know if in cocoa this is also the behavior when using bindings. If you bind the content binding to an array containing @"" and select it, does it automatically set the selected index to -1 ?

**Comment by daboe01** [comment]
cocoa inserts empty strings into the item lists.
bildschirmfoto 2018-03-31 um 17 32 46

```ObjC
    NSMutableDictionary *value = [[NSMutableDictionary alloc] init];
    [value setObject:[NSString stringWithFormat:@""] forKey:@"name"];
    [self.myAC addObject:value];
    [self.myAC rearrangeObjects];
```

File: AppKit/CPPopUpButton.j [hunk 1]
```diff
@@ -491,9 +491,17 @@ CPPopUpButtonStatePullsDown = CPThemeState("pulls-down");
 
         if (index < 0)
         {
-            [self addItemWithTitle:aTitle];
 
-            index = [self numberOfItems] - 1;
+            // this ist to match cocoa where setting an empty string does not add it but simply clears the title
+            // and sets objectValue to -1
+            if (aTitle === '')
+                [self selectItemAtIndex:-1];
```
Comment (cacaodev) [inline AppKit/CPPopUpButton.j]: What about `index = -1` ?

Comment (daboe01) [reply AppKit/CPPopUpButton.j]: i am not sure i understand you question 100%.

index = -1 means "this popupbutton has currently no selected item".

Comment (daboe01) [reply AppKit/CPPopUpButton.j]: i have to check tonight the value in cocoa with bindings.

**Comment by daboe01** [comment]
-#new
+AppKit
milestone=1.0

File: AppKit/CPPopUpButton.j (hunk 1 above)
Comment (daboe01) [reply AppKit/CPPopUpButton.j]: @cacaodev the object value seems to equal the selected index in cocoa when the popupbutton is filled via bindings, regardless whether the title is the empty string.

**Pull request merged**